pacfetcher.go

```diff
@@ -80,6 +83,46 @@ func requireOK(resp *http.Response, err error) (*http.Response, error) {
 	}
 }
 
+// decodeDataURL decodes a data URL (e.g., data:text/plain;base64,SGVsbG8sIFdvcmxkIQ==).
+// It supports both base64 and URL-encoded data, and enforces the maxResponseBytes size limit.
+// See https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs for details.
+func decodeDataURL(uri string) ([]byte, error) {
+	parsed_url, err := url.Parse(uri)
+
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing pac url: %w", err)
+	}
+
+	if parsed_url.Scheme != "data" {
```

Comment:
we should call this `parsedURL` as per https://go.dev/doc/effective_go#mixed-caps

```suggestion
	parsedURL, err := url.Parse(uri)

	if err != nil {
		return nil, fmt.Errorf("Error parsing pac url: %w", err)
	}

	if parsedURL.Scheme != "data" {
```